packages/lab/src/components/Flow/Node/Node.tsx

```diff
@@ -0,0 +1,192 @@
+import { useFlowContext } from "../FlowContext/FlowContext";
+
+import { HvFlowNodeInput, HvFlowNodeOutput, HvFlowNodeParam } from "../types";
+import { staticClasses, useClasses } from "./Node.styles";
+import ParamRenderer from "./Parameters/ParamRenderer";
+
+export { staticClasses as flowNodeClasses };
+
+export type HvFlowNodeClasses = ExtractNames<typeof useClasses>;
+
+export interface HvFlowNodeProps extends Omit<HvBaseProps, "id">, NodeProps {
+  /** Node title */
+  title: string;
+  /** Node description */
+  description: string;
+  /** Node color */
+  color?: HvColorAny;
+  /** Node expanded */
+  expanded?: boolean;
+  /** Node inputs */
+  inputs?: HvFlowNodeInput[];
+  /** Node outputs */
+  outputs?: HvFlowNodeOutput[];
+  /** Node parameters */
+  params?: HvFlowNodeParam[];
+  /** A Jss Object used to override or extend the styles applied to the component. */
+  classes?: HvFlowNodeClasses;
+}
+
+const isInputConnected = (id, type, idx, edges) => {
+  if (type === "target") {
+    return edges.some(
+      (e) => e.target === id && e.targetHandle === idx.toString()
+    );
+  }
+  if (type === "source") {
+    return edges.some(
+      (e) => e.source === id && e.sourceHandle === idx.toString()
+    );
+  }
+};
+
+export const HvFlowNode = ({
+  id,
+  type,
+  title,
+  description,
+  color: colorProp = theme.colors.cat3_40,
+  expanded = false,
+  params,
+  classes: classesProp,
+  className,
+}: HvFlowNodeProps) => {
+  const [showParams, setShowParams] = useState(expanded);
+  const reactFlowInstance = useReactFlow();
+
+  const { classes, cx, css } = useClasses(classesProp);
+  const color = getColor(colorProp);
+
+  const { nodeGroups, nodeTypes } = useFlowContext();
+  const edges = useStore((s) => s.edges);
+  const nodes = useStore((s) => s.getNodes());
+
+  const node = nodes.find((n) => n.id === id);
+
+  const groupId = nodeTypes?.[type].meta?.groupId;
+  const inputs = nodeTypes?.[type]?.meta?.inputs;
+  const outputs = nodeTypes?.[type]?.meta?.outputs;
+  const icon = groupId && nodeGroups && nodeGroups[groupId].icon;
+
+  useEffect(() => {
+    const newNodes = nodes.map((n) => {
+      if (n.id === id) {
+        if (Object.keys(n.data).length === 0) {
+          params?.forEach((param) => {
+            n.data[param.label] = param.value;
+          });
+        }
+      }
+      return n;
+    });
+    reactFlowInstance.setNodes(newNodes);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const hasParams = !!(params && params.length > 0);
+
+  return (
+    <div
+      className={cx(
+        css({ border: `1px solid ${color}` }),
+        classes.root,
+        className
+      )}
+    >
+      <div
+        className={cx(css({ backgroundColor: color }), classes.headerContainer)}
+      >
+        <div className={classes.titleContainer}>
+          {icon}
+          <HvTypography variant="title4" className={classes.title}>
+            {groupId}
```

Comment:
Instead of using the `groupId` here, I think we should use the group `label` (`nodeTypes?.[type].meta?.label`). Using an `id` seems confusing specially if someone is not careful about the ids they are providing to the groups.
Also, I think we should rename the `titleContainer` class used here. It can be confusing since we have a `title` prop that we use below. 
What do you think?